.github/workflows/typescript.yml

```diff
@@ -19,4 +19,6 @@ jobs:
       - name: Install dependencies
         run: npm install
       - name: Run TypeScript compiler
-        run: npm exec tsc -- --noEmit
+        run: npm run typecheck
+      - name: Run Eslint checks
```

Comment:
ESLint can be a separate workflow. This gives better visibility on the PR page which check failed. And as the project grows, will allow checks to run faster since they'll run in parallel. 

There's an argument for joining them to save CI minutes (since the setup is duplicated), but I think it's unlikely we'll exceed the free tier minutes anyway.

Reply:
i don't think its necessary 
the developer should run this tests locally
the workflow is just for blocking the merge if the test fails

Reply:
typescript and eslint will never take long enough to be a problem
we should just run other workflow if we implement a actual testing library
that would take time enough to be a problem

Reply:
On a project of this size, that's a reasonable stance. I'm a little biased by experience on larger projects where ESLint runtime can be problematic.

I'm fine with keeping it as one workflow. But it should renamed to something more generic, since currently it's called "Typescript Compilation Check", which doesn't indicate that it will also fail on ESLint errors. Perhaps we could use "Static Analysis" as a blanket term for the lint and type checks.